### PR TITLE
[sqlite] custom sqlite and sqlcipher

### PR DESCRIPTION
**Pull request by Kudo**
# Why

close ENG-11976
close ENG-12401

# How

- to fix the incompatible symbol with system sqlite on ios, we now replace `sqlite3_*` api as `exsqlite3_*`.
- we then have to store sqlite3 source code on our git repo as well as publish to npm. in theory we can download sqlite and replace symbols on build time, but then we will have difficulty to run `./configure` on windows to sqlite3.c building. publishing npm would be easier.
- sqlcipher is then feasible.

the **scripts/prepare_sqlite.ts** is for downloading sqlite and sqlcipher. the **scripts/replace_symbols.ts** is to replace `exsqlite3_` in both sqlite, sqlcipher and our source code. both these scripts are for maintenance when we need to upgrade sqlite version.  

the expo-sqlite tarball size will increase from 2.9MB to 7.7MB

# Test Plan

- ci passed
- add sqlcipher test case in test-suite and add `expo.sqlite.enableFTS` to **ios/Podfile.properties.json` / **android/gradle.properties** to bare-expo for testing.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by Kudo** [comment]
* **#30824** <a href="https://app.graphite.dev/github/pr/expo/expo/30824?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Kudo and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/expo-sqlite/android/build.gradle [hunk 1]
```diff
@@ -79,41 +76,6 @@ android {
 }
 
 dependencies {
+  compileOnly 'com.android.ndk.thirdparty:openssl:1.1.1q-beta-1'
```
Comment (Kudo) [inline packages/expo-sqlite/android/build.gradle:79]:
```suggestion
  if (project.ext.USE_SQLCIPHER) {
    compileOnly 'com.android.ndk.thirdparty:openssl:1.1.1q-beta-1'
  }
```

File: packages/expo-sqlite/ios/CRSQLiteLoader.m [hunk 2]
```diff
@@ -1,17 +1,17 @@
 // Copyright 2015-present 650 Industries. All rights reserved.
 
 #import <ExpoSQLite/CRSQLiteLoader.h>
-#import <sqlite3/sqlite3.h>
+#import <ExpoSQLite/sqlite3.h>
 
 int crsqlite_init_from_swift(sqlite3 *db) {
-  sqlite3_enable_load_extension(db, 1);
+  exsqlite3_enable_load_extension(db, 1);
```
Comment (alanjhughes) [inline packages/expo-sqlite/ios/CRSQLiteLoader.m:7]: This reminds me. We should remove crsqlite. It's no longer maintained. Matt has moved on to other things

**Review by alanjhughes — APPROVED** [review]

File: packages/expo-sqlite/ios/CRSQLiteLoader.m (hunk 2 above)
Comment (Kudo) [reply packages/expo-sqlite/ios/CRSQLiteLoader.m:7]: yes, we've talked this with Brent. we would remove it from sdk 53 and give a deprecated warning in sdk 52
https://linear.app/expo/issue/ENG-12984/deprecate-enablecrsqlite-and-warn-when-used

**Comment by Kudo** [comment]
### Merge activity

* **Aug 8, 10:54 PM EDT**: @Kudo started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/expo/expo/30824).

**Pull request merged**